docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,186 @@
+
+```js
+t( 'insert emoji' );
+t( 'insert %0 emoji', emojiName );
+t( 'insert %0 emoji', [ emojiName ] );
+```
+
+For more advanced scenarios use plain objects forms:
+
+```js
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' );
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity );
+t( { string: '%1 %0 emoji', plural: '%1 %0 emojis', id: 'INSERT_EMOJIS' }, [ quantity, 'Insert' ] );
+```
+
+### Examples
+
+#### Localizing UI of plugins:
+
+```js
+// ...
+editor.ui.componentFactory.add( 'smilingFaceEmoji', locale => {
+	const command = editor.commands.get( 'insertSmilingFaceEmoji' );
+	const buttonView = new ButtonView( locale );
+
+	// The translation function.
+	const { t } = editor.locale;
+
+  // The localized label.
+	const label = t( 'insert smiling face emoji' );
+
+	buttonView.set( {
+		label,
+		icon: emojiIcon,
+		tooltip: true
+	} );
+
+	buttonView.on( 'execute', () => {
+		editor.execute( 'insertSmilingFaceEmoji' );
+		editor.editing.view.focus();
+	} );
+} );
+// ...
+```
+
+Note that this sample lacks a few parts. To check how to create a complete plugin, for example, check the {@link framework/guides/creating-simple-plugin Creating a simple plugin guide}.
+
+#### Localizing aria attributes:
+
+```js
+editingView.change( writer => {
+	const editingView = this._editingView;
+	const t = this.t;
+
+	editingView.change( writer => {
+		const viewRoot = editingView.document.getRoot( this.name );
+
+		writer.setAttribute( 'aria-label', t( 'Rich Text Editor, %0', [ this.name ] ), viewRoot );
+	} );
+} );
+```
+
+#### Localizing pending actions
+
+Pending actions are used to inform the user that the action is in progress and we will lost data while exiting the editor - see {@link module:core/pendingactions~PendingActions the `PendingActions` class}
+
+```js
+_updatePendingAction() {
+	// ...
+	const pendingActions = this.editor.plugins.get( PendingActions );
+
+	// ...
+	const t = this.editor.t;
+	const getMessage = value => `${ t( 'Upload in progress' ) } ${ parseInt( value ) }%.`;
+
+	this._pendingAction = pendingActions.add( getMessage( this.uploadedPercent ) );
+	this._pendingAction.bind( 'message' ).to( this, 'uploadedPercent', getMessage );
+}
+```
+
+## Adding translations and localizing the editor UI
+
+First of all, if you have found a missing or an incorrect translation in the CKEditor 5 translations, {@link framework/guides/contributing/contributing#translating check how you can contribute to the project}! Your help will be appreciated by others!
+
+Adding translations to the editor can be done in three ways to satisfy various needs.
+
+- by adding translations via {@link module:utils/translation-service.add the translation-service's `add()` function} - this function can be used at runtime before the editor starts
+- by extending the global `window.CKEDITOR_TRANSLATIONS` object - this object can be modified before and after the build step. CKEditor 5 translations assets operate on this object.
```

Comment:
"this object can be modified before and after the build" - well this method can also be simply used at a runtime, isn't it? In exactly the same way as the first point (with the exception that the first needs an editor instance access.

If that's the case I think same wording should be used so keyword runtime should be used.

Reply:
Maybe `runtime` isn't the best wording here, but I improved the description.